batchflow/models/torch/base.py

```diff
@@ -736,44 +653,116 @@ def make_decay(self, decay, optimizer=None, **kwargs):
             self.decay_step.append(step_params)
 
 
-    # Use an external model
+    # Set pre-initialized model or chain multiple building blocks to create model
     def set_model(self, model):
-        """ Set the underlying model to a supplied one and update training infrastructure. """
+        """ Set the underlying PyTorch model to a supplied one and update training infrastructure. """
         self.model = model
         self.initialize_weights()
+        self.model_to_device()
 
+        self.make_infrastructure()
 
-        self._to_device()
 
-        self.make_loss(**self.unpack('loss'))
-        self.make_optimizer(**self.unpack('optimizer'))
-        self.make_decay(**self.unpack('decay'), optimizer=self.optimizer)
-        self.scaler = torch.cuda.amp.GradScaler()
+    def build_model(self, inputs=None):
+        """ Create the instance of PyTorch model by chaining multiple blocks sequentially.
+        After it, create training infrastructure (loss, optimizer, decay).
+
+        The order is defined by `order` key in the config, which is [`initial_block`, `body`, `head`] by default.
+        Each item in `order` should describe the block name, the config name and method to create. It can be a:
+            - string, then we use at as name, key and method name
+            - tuple of three elements, which are name, key and method name or callable
+            - dictionary with three items, which are `block_name`, `config_name` and `method`.
+
+        The `block_name` is then used as the identifier in resulting model, i.e. `model.initial_block`, `model.body`.
+        The `config_name` is used to retrieve block creation parameters from config.
+        The `method` is either a string to getattr from the current instance or a callable.
+        Either method or callable should return an instance of nn.Module and accept block parameters.
```

Comment:
indent this block to match the dictionary description above

Last line is not quite clear: what do you refer to by 'method'

Reply:
"name of the method to get from the current instance" - this description from below is much more clear